### PR TITLE
Merge4wd features to magni

### PR DESCRIPTION
**Pull request by mjstn**
This is a syncing up of quite a few features that have been enhanced or added to the 4WD platform so the codebase for that ubiquity_motor is just about like that of the standard Magni 2 wheel drive code for ubiquity_motor.

Several features that would not be active for Magni 2WD are in this code but would normally be off for a Magni.

There is added a concept of 'drive_type' that comes in that would be  2wd for magni.
There are quite a few more  'pid control' bits defined that again would normally be off for a standard Magni.

Other useful addition is we will publish to a MotorState topic now and show wheel PWM and current in use at any given time.  This state was valuable for our 4WD product where the wheels must skid to steer but for magni it is simply another diagnostic.

### CODE REVIEW

**Comment by mjstn** [comment]
I am in a pickle on selection of the reviewer.  I would like it to be Vid or Janez.  Vid is out of here soon and already overworked to finish things before he takes off and Janez is not yet back and almost certain to be super busy with accumulated work.

ONLY ONE OR OTHER reviewer is needed.   If one person takes it, say so in a comment so other will know it is 'taken', thanks.

The 'rush' on this PR is that this has to go in and be merged prior to ANOTHER planned PR to support 5.17 gear ratio wheels.  I do not want that PR to be folded into this merge, I want it to be separate so this PR is 'A Little Bit Hot' but not 'Liquid Magma' hot.

**Comment by JanezCim** [comment]
I can take it. But I do not have a robot with me, I only have a RPI, touchscreen and MCB. Can I test this with these parts? If yes, please give me a list of stuff that I need to do. If the whole robot is needed for testing this you will need to assign it so someone else.

**Comment by mjstn** [comment]
Thanks!   You do not have to run or compile any code for a review.     I have myself been testing it and actively today  Timotej is testing the code and also some new firmware.    So this review is to look at the code and look for stupid mistakes or things that seem odd.    A code review is not normally including active testing or verification that it makes, that is assumed to be done or in progress by myself and the test group which it is in fact the case.

File: CMakeLists.txt [hunk 1]
```diff
@@ -7,7 +7,9 @@ set(CMAKE_CXX_STANDARD 14) # use C++14
 ## if COMPONENTS list like find_package(catkin REQUIRED COMPONENTS xyz)
 ## is used, also find other catkin packages
 find_package(catkin REQUIRED COMPONENTS 
+	message_generation
 	roscpp 
+        rospy
```
Comment (JanezCim) [inline CMakeLists.txt:12]: 2 spaces instead of a tab maybe here

File: include/ubiquity_motor/motor_message.h [hunk 2]
```diff
@@ -88,7 +91,8 @@ class MotorMessage {
         REG_PWM_BOTH_WHLS    = 0x07,  // ReadOnly: Both Motor PWM packed as two 16-bit ints. M1 upper word
         REG_TINT_BOTH_WHLS   = 0x08,  // ReadOnly: Both Motor Tic Interval packed as two 16-bit ints. M1 upper word
 
-        REG_LEFT_RAMP        = 0x09,  // Deprecated
+        REG_09               = 0x09,  // Deprecated
+	REG_DRIVE_TYPE       = 0x0A,  // The type of wheel-motor in use                            [v42]
```
Comment (JanezCim) [inline include/ubiquity_motor/motor_message.h]: tab instead of 2 spaces?

File: src/motor_node.cc [hunk 3]
```diff
@@ -189,6 +224,16 @@ void initMcbParameters(std::unique_ptr<MotorHardware> &robot )
         mcbStatusPeriodSec.sleep();
     }
 
+    // Certain 4WD robots rely on wheels to skid to reach final positions.
+    // For such robots when loaded down the wheels can get in a state where they cannot skid.
+    // This leads to motor overheating.  This code below sacrifices accurate odometry which
+    // in not achievable in such robots anyway to relieve high wattage drive power when zero velocity.
```
Comment (JanezCim) [inline src/motor_node.cc]: **is** not achievable

**Review by JanezCim — APPROVED** [review]
seems ok to me visually, havent tested on a robot. The only comment is that on a lot of places spaces and tabs are not formatted properly (i have not commented on all of them). Probably your IDE is formatting ok and then on github it gets missaligned. Otherwise i dont see any stupid mistakes :)

**Comment by mjstn** [comment]
Fixed tabstops and one simple typo in a comment per review request. Thanks Janez.

**Pull request merged**